tests/Installer/PackagesInstallerTest.php

```diff
@@ -11,121 +11,184 @@
+                    };
+                },
+            );
+
+        $installer->install();
+    }
+
+    public function testCanMergeRecursivelyWhenInstalling(): void
+    {
+        $composer     = $this->createMock(Composer::class);
+        $typeResolver = $this->createMock(ProjectTypeResolver::class);
+        $depInstaller = $this->createMock(DependencyInstaller::class);
+        $io           = $this->createMock(IOInterface::class);
+
+        $mapping = [
+            MappingResolver::DEFAULT_MAPPING_TYPE => [
+                'phpunit/phpunit' => [
+                    'version' => '@stable',
+                    'dev' => true,
+                    'updateDependencies' => true,
+                    'allowVersionOverride' => false,
+                ],
             ],
-            [
-                'default',
-                $this->createLinkMocks(['foo/bar']),
-                null
+            'magento2' => [
+                'phpunit/phpunit' => [
+                    'version' => '^10.6.5',
+                    'dev' => false,
+                    'allowVersionOverride' => true,
+                ],
             ],
-            [
-                'unknown',
-                $this->createLinkMocks(['foo/bar']),
-                null
-            ]
         ];
+
+        $typeResolver
+            ->method('resolve')
+            ->willReturn('magento2');
+
+        $installer = new PackagesInstaller(
+            $composer,
+            $typeResolver,
+            $io,
+            $depInstaller,
+            $mapping,
+        );
+
+        $depInstaller
+            ->expects($this->exactly(1))
+            ->method('installPackage')
+            ->with('phpunit/phpunit', '^10.6.5', false, true, true);
+
+        $installer->install();
     }
 
-    /**
-     * @param string[] $targets
-     *
-     * @return Link[]
-     */
-    private function createLinkMocks(array $targets): array
+    public function testPhpUnitIsInstalledForUnknownProjectType(): void
     {
-        return array_map(
-            function (string $target): Link {
-                /** @var Link $mock */
-                $mock = $this->createConfiguredMock(
-                    Link::class,
-                    ['getTarget' => $target]
-                );
-
-                return $mock;
-            },
-            $targets
+        $composer     = $this->createMock(Composer::class);
+        $typeResolver = $this->createMock(ProjectTypeResolver::class);
+        $depInstaller = $this->createMock(DependencyInstaller::class);
+        $io           = $this->createMock(IOInterface::class);
+
+        $typeResolver
+            ->method('resolve')
+            ->willReturn('foobar');
+
+        $installer = new PackagesInstaller(
+            $composer,
+            $typeResolver,
+            $io,
+            $depInstaller,
         );
+
+        $depInstaller
+            ->expects($this->exactly(1))
+            ->method('installPackage');
```

Comment:
Please add a `->with()` here too -- that the correct package is installed is more relevant then that the correct information message is written to io ;)